File: web/src/api/gophish.js

/**
 * 社会工程 钓鱼活动
 */

import axios from '@/axios/http';

const gophish = {
    campaignall(param){
        return axios.get('/smart/gophish/campaign/all',param) 
    },
    campaigncreate(param){
        return axios.postJson('/smart/gophish/campaign/create',param) 
    },
    campaignresult(param){
        return axios.get('/smart/gophish/campaign/result',param) 
    },
    campaigndetail(param){
        return axios.get('/smart/gophish/campaign/detail',param) 
    },
    campaigncomplete(param){
        return axios.postJson('/smart/gophish/campaign/complete',param) 
    },
    campaigndelete(param){
        return axios.postJson('/smart/gophish/campaign/delete',param) 
    },
    // 目标组
    groupall(param){
        return axios.get('/smart/gophish/group/all',param) 
    },
    groupcreate(param){
        return axios.postJson('/smart/gophish/group/create',param) 
    },
    groupupdate(param){
        return axios.postJson('/smart/gophish/group/update',param) 
    },
    groupdetail(param){
        return axios.get('/smart/gophish/group/detail',param) 
    },
    groupdelete(param){
        return axios.postJson('/smart/gophish/group/delete',param) 
    },
    //模板
    templateall(param){
        return axios.get('/smart/gophish/template/all',param) 
    },
    templatecreate(param){
        return axios.postJson('/smart/gophish/template/create',param) 
    },
    templateupdate(param){
        return axios.postJson('/smart/gophish/template/update',param) 
    },
    templatedelete(param){
        return axios.postJson('/smart/gophish/template/delete',param) 
    },
    // 钓鱼网站
    pageall(param){
        return axios.get('/smart/gophish/page/all',param) 
    },
    pagecreate(param){
        return axios.postJson('/smart/gophish/page/create',param) 
    },
    pageupdate(param){
        return axios.postJson('/smart/gophish/page/update',param) 
    },
    pagedelete(param){
        return axios.postJson('/smart/gophish/page/delete',param) 
    },
    pageimport_site(param){
        return axios.postJson('/smart/gophish/page/import_site',param) 
    },
    //发件配置
    profileall(param){
        return axios.get('/smart/gophish/profile/all',param) 
    },
    profilecreate(param){
        return axios.postJson('/smart/gophish/profile/create',param) 
    },
    profileupdate(param){
        return axios.postJson('/smart/gophish/profile/update',param) 
    },
    profiledelete(param){
        return axios.postJson('/smart/gophish/profile/delete',param) 
    },
    send_test_email(param){
        return axios.postJson('/smart/gophish/profile/send_test_email',param) 
    },
}

export default gophish;

File: routers/gophish.go
package routers

import (
	"smart/api/rest/gofish"

	"github.com/gin-gonic/gin"
)

func GoPhishRouter(r *gin.RouterGroup) {
	// 模块前缀
	group := r.Group("/gophish")
	api := gofish.GoPhishAPI{}

	group.GET("/group/all", api.GetGroupAll)
	group.GET("/group/detail", api.GetGroupDetail)
	group.POST("/group/create", api.GroupCreate)
	group.POST("/group/update", api.GroupUpdate)
	group.POST("/group/delete", api.GroupDelete)

	// templates
	group.GET("/template/all", api.TemplateAll)
	group.GET("/template/detail", api.TemplateDetail)
	group.POST("/template/create", api.TemplateCreate)
	group.POST("/template/update", api.TemplateUpdate)
	group.POST("/template/delete", api.TemplateDelete)
	group.POST("/template/import_email", api.TemplateImportEmail)

	// landing pages
	group.GET("/page/all", api.PageAll)
	group.GET("/page/detail", api.PageDetail)
	group.POST("/page/create", api.PageCreate)
	group.POST("/page/update", api.PageUpdate)
	group.POST("/page/delete", api.PageDelete)
	group.POST("/page/import_site", api.PageImportSite)

	// profiles (smtp)
	group.GET("/profile/all", api.ProfileAll)
	group.GET("/profile/detail", api.ProfileDetail)
	group.POST("/profile/create", api.ProfileCreate)
	group.POST("/profile/update", api.ProfileUpdate)
	group.POST("/profile/delete", api.ProfileDelete)
	group.POST("/profile/send_test_email", api.ProfileSendTestEmail)

	// campaigns
	group.GET("/campaign/all", api.CampaignAll)
	group.GET("/campaign/detail", api.CampaignDetail)
	group.POST("/campaign/create", api.CampaignCreate)
	group.POST("/campaign/update", api.CampaignUpdate)
	group.POST("/campaign/delete", api.CampaignDelete)
	group.POST("/campaign/launch", api.CampaignLaunch)
	group.POST("/campaign/complete", api.CampaignComplete)
	group.GET("/campaign/result", api.CampaignResult)
}
